src/components/Publication/Publication.js

```diff
@@ -75,22 +75,29 @@ class Publication extends Component {
       subDoc: '',
     };
 
-    this.arethusa = new ArethusaWrapper();
-    this.arethusaSubDocFun = this.arethusaSubDocFun.bind(this);
-  }
+    this.setSubdoc = this.setSubdoc.bind(this);
 
-  componentDidMount() {
+    this.arethusa = new ArethusaWrapper();
     // eslint-disable-next-line no-undef
-    window.arethusaSubDocFun = this.arethusaSubDocFun;
+    this.interval = window.setInterval(this.setSubdoc, 100);
```

Comment:
I don't particularly like the idea of having a timer in the component, but in the end this was the only solution I could get to work. The issue is that `widget.api().getSubdoc()` throws an error (or sometimes returns `undefined`) until Arethusa has fully loaded.

I'm open to any suggestions!

Reply:
I think there might be some way to get Arethusa to tell us when it's fully loaded but I have to investigate the code to find out. Maybe you could enter an issue on Arethusa and assign it to me and then add one to the treebank template to get rid of the use of the timeout once that is fixed? I don't know when I will get to it, so I don't want to hold up the rest of the progress for it.

Reply:
Great, I've created a new issue here: https://github.com/alpheios-project/arethusa/issues/793